packages/terminal/src/widget.ts

```diff
@@ -400,4 +427,39 @@ namespace Private {
     cursorAccent: '#000',
     selection: 'rgba(255, 255, 255, 0.3)'
   };
+
+  /**
+   * A Promise for the initial load of xterm.
+   */
+  export let xtermReady: Promise<void>;
+
+  /**
+   * Lazy-load the xterm library and addons
+   */
+  export function ensureXTerm(): Promise<void> {
+    if (xtermReady) {
+      return xtermReady;
+    }
+
+    xtermReady = new Promise((resolve, reject) => {
+      require.ensure(
```

Comment:
In jupyterlab/jupyterlab-latex#107 we used ES6 dynamic imports here instead of `require.ensure`. I don't think it matters much to webpack, but the dynamic import seems more future-proof. Are you aware of any practical differences between the options?

Reply:
`esnext` exports break `ts-jest`, last i checked. I would be heartily :100: getting everything on `esnext` before the real `1.0`, but doing it piece-wise is probably no good. Since I'd have to fix the terminal tests _anyway_, i'll take a look if some combination of bumping versions and black magic makes it all magically work.

Reply:
Makes sense, no rush to do it here.